javacutil/src/org/checkerframework/javacutil/TreeUtils.java

```diff
@@ -355,6 +355,29 @@ public static ExpressionTree skipParens(final ExpressionTree tree) {
         return t;
     }
 
+    /**
+     * If the given tree is a block tree with a single element, return the enclosed non-block
+     * statement. Otherwise, return the same tree.
+     *
+     * @param tree a statement tree
+     * @return the single enclosed statement, if it exists; otherwise, the same tree
+     */
+    public static StatementTree skipBlocks(final StatementTree tree) {
```

Comment:
This method is only needed because the OptionalVisitor operates over Trees instead of over the CFG. I would rather move this to a private helper method in OptionalVisitor. A checker shouldn't write code like this.

Reply:
Does the same comment apply to skipParens and possibly other methods in TreeUtils?

Reply:
Possibly for `skipParens`, yes. There are very few uses of that method and some in InterningVisitor might be better implemented over the CFG.
`skipParens` would still be useful to print shorter error messages and for some Tree-based checks. I don't think we ever need to do that with blocks.